### PR TITLE
Clustering and classification

### PR DESCRIPTION
**Pull request by jorana**
**What is the problem / what does the code in this PR do**
Requires, https://github.com/AxFoundation/strax/pull/403.

For motivation, please refer to the notes linked here:
https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenonnt:analysis:waveforms_team:20210408

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 strax>=0.14.0
-utilix>=0.4.1
+utilix==0.4.1
```
Comment (jorana) [inline requirements.txt:2]: Temporary until https://github.com/XENONnT/utilix/pull/41 is fixed

File: straxen/contexts.py [hunk 2]
```diff
@@ -255,26 +255,34 @@ def xenonnt_initial_commissioning(*args, **kwargs):
...
         ((0.5, 1), (3.5, 0.25)),
         ((2, 1), (4.5, 0.4))),
+    peak_min_pmts=2,
+    # MergedS2s
+    s2_merge_max_duration=5000.,
+    s2_merge_max_gap=3500,
+    # Peaks
+    # Smaller right extension since we applied the filter
+    peak_right_extension=30,
+    # Events*
```
Comment (jorana) [inline straxen/contexts.py:285]: While adding these 1T configs, I noticed that they were very disorganized. Therefore I slightly re-ordered the configs.

**Comment by skazama** [comment]
Hi @jorana, thanks for the PR. I just checked the code and have a question:
It seems that you have updated value for s2_merge_max_area to 35_000, but this should be for s2_merge_max_duration, right?
Maybe we also need to check which value we should use for s2_merge_max_area, but at least 35_000 is for s2_merge_max_duration if I understand correctly.

**Comment by jorana** [comment]
hi @skazama , thanks for the careful check, you are absolutely right. Too many very similar options 🤔 . 

I had checked `s2_merge_max_area` as well in my studies, but did not find evidence of a need to increase this number.

**Review by skazama — APPROVED** [review]
Thanks @jorana for the update. From my side, It looks fine now, but maybe let's wait @zhut19 and @jhowl01's responses.

File: straxen/plugins/peaklet_processing.py [hunk 3]
```diff
@@ -26,8 +26,8 @@
                  # for more information
                  default=(
                      None,  # Reserved
-                     ((0.5, 1), (4, 0.4)),
-                     ((2, 1), (4.5, 0.4))),
+                     ((0.5, 1.), (6.0, 0.4)),
```
Comment (zhut19) [inline straxen/plugins/peaklet_processing.py:29]: This is a typo, right?

Comment (zhut19) [reply straxen/plugins/peaklet_processing.py:29]: Sorry, I missed that plot.

Comment (skazama) [reply straxen/plugins/peaklet_processing.py:29]: @zhut19 do you think this PR can be merged now?

**Review by zhut19 — APPROVED** [review]

**Review by jhowl01 — APPROVED** [review]
Looks great! Excellent work @jorana !

My only thing is it seems like we're making these changes to the defaults and then overwriting them for outdated contexts. Wouldn't it be better to explicitly set at least the clustering parameters that are likely to change several times inside `contexts.py`? Maybe in `xnt_common_config`? This is probably a matter of taste but I remember ages of checking the git blame on XENON1T.ini in pax, and in that case it was pretty useful to find all the changes in one place.

Approving though because again, I think this is just one guy's opinion

**Comment by jorana** [comment]
> 
> 
> Looks great! Excellent work @jorana !
> 
> My only thing is it seems like we're making these changes to the defaults and then overwriting them for outdated contexts. Wouldn't it be better to explicitly set at least the clustering parameters that are likely to change several times inside `contexts.py`? Maybe in `xnt_common_config`? This is probably a matter of taste but I remember ages of checking the git blame on XENON1T.ini in pax, and in that case it was pretty useful to find all the changes in one place.
> 
> Approving though because again, I think this is just one guy's opinion

Hi Joey, thanks for asking! The convention we have in straxen is to update the default of the plugins-option if it relates to xenon-nt to increase the readability of the plugins. There are several examples where we don't really follow this convention due to https://github.com/AxFoundation/strax/issues/400. Then for xenon1t (which at this point shouldn't change much), we set the values in the config of the context.

**Pull request merged**